testdata/ecosystem/pipelines/cache-stepactions-python.yaml

```diff
@@ -0,0 +1,108 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: demo-pipeline-tekton-caches-python
+spec:
+  params:
+    - name: repo_url
+      type: string
+    - name: revision
+      type: string
+      default: main
+    - name: registry
+      type: string
+    - name: cachePatterns
+      type: array
+      default: ["**requirements.txt"]
+    - name: image
+      type: string
+      default: python:latest
+    - name: force-cache-upload
+      type: string
+      default: "false"
+  workspaces:
+    - name: source
+  tasks:
+    - displayName: Build Python Application
+      name: build-task
+      workspaces:
+        - name: source
+          workspace: source
+      taskSpec:
+        workspaces:
+          - name: source
+        params:
+          - name: cachePatterns
+            default: $(params.cachePatterns)
+          - name: image
+            default: $(params.image)
+        steps:
+          - name: create-repo
+            image: $(params.image)
+            script: |
+              mkdir -p $(workspaces.source.path)/repo
+              chmod 777 $(workspaces.source.path)/repo
+          - name: clone-repo
```

Comment:
Please use `git-clone` step action that we ship in namespace `openshift-pipelines` + cluster resolver.